===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_asn1_read:value)) at 12 (line 219, column 2) }
===== Actual patch in diff format =====
@@ -216,7 +216,11 @@ p11_asn1_read (node_asn *asn,
 	return_val_if_fail (value != NULL, NULL);
 
 	ret = asn1_read_value (asn, field, value, &len);
-	return_val_if_fail (ret == ASN1_SUCCESS, NULL);
+	if (ret == ASN1_SUCCESS) { } else {
+		p11_debug_precond ("p11-kit: '%s' not true at %s\n", "ret == ASN1_SUCCESS", __func__);
+		free(value);
+		return NULL;
+	}
 
 	/* Courtesy zero terminated */
 	value[len] = '\0';
